commit 7d6555b11f181bdc24ba56577f753a07add6e8c7
Author: Matt Jolly <kangie@gentoo.org>
Date:   Tue Jun 16 21:57:07 2026 -0700

    build: Fix get_path_info on empty ar in unbundle toolchain
    
    Some toolchains leave ar empty during initial setup, causing GN to error
    when get_path_info() is called with an empty string. Guard the check to
    only run when ar is not empty.
    
    Signed-off-by: Matt Jolly <kangie@gentoo.org>
    Change-Id: I87615806ddfda6f262a7500b0c5b6fed1f452985
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7949777
    Reviewed-by: Takuto Ikuta <tikuta@chromium.org>
    Commit-Queue: Takuto Ikuta <tikuta@chromium.org>
    Reviewed-by: Matt Stark <msta@google.com>
    Cr-Commit-Position: refs/heads/main@{#1648076}

--- a/build/toolchain/gcc_toolchain.gni
+++ b/build/toolchain/gcc_toolchain.gni
@@ -408,8 +408,9 @@ template("single_gcc_toolchain") {
       } else {
         command = "rm -f {{output}} && $command"
 
-        # Add ar to inputs if it's not from $PATH.
-        if (get_path_info(ar, "file") != ar) {
+        # Add ar to inputs if it's not from $PATH. Some toolchains leave
+        # |ar| empty during toolchain setup, so guard get_path_info() here.
+        if (ar != "" && get_path_info(ar, "file") != ar) {
           inputs =
               [ get_path_info(rebase_path(ar, ".", root_out_dir), "abspath") ]
         }
